aiida/tools/importexport/dbimport/backends/django.py

```diff
@@ -0,0 +1,689 @@
+        batch_size: int = get_config_option('db.batch_size')
+
+        with transaction.atomic():
+
+            # entity_name -> str(pk) -> fields
+            new_entries: Dict[str, Dict[str, dict]] = {}
+            existing_entries: Dict[str, Dict[str, dict]] = {}
+            # entity_name -> identifier -> pk
+            foreign_ids_reverse_mappings: Dict[str, Dict[str, int]] = {}
+
+            IMPORT_LOGGER.debug('ASSESSING IMPORT DATA...')
+            for entity_name in entity_order:
+                _generate_entity_data(
+                    entity_name=entity_name,
+                    reader=reader,
+                    new_entries=new_entries,
+                    existing_entries=existing_entries,
+                    foreign_ids_reverse_mappings=foreign_ids_reverse_mappings,
+                    extras_mode_new=extras_mode_new,
+                )
+
+            IMPORT_LOGGER.debug('STORING ENTITIES...')
+            for entity_name in entity_order:
+                _store_entity_data(
+                    reader=reader,
+                    entity_name=entity_name,
+                    comment_mode=comment_mode,
+                    extras_mode_existing=extras_mode_existing,
+                    new_entries=new_entries,
+                    existing_entries=existing_entries,
+                    foreign_ids_reverse_mappings=foreign_ids_reverse_mappings,
+                    import_unique_ids_mappings=import_unique_ids_mappings,
+                    ret_dict=ret_dict,
+                    batch_size=batch_size,
+                    # session=session
+                )
+
+            IMPORT_LOGGER.debug('STORING NODE LINKS...')
+            _store_node_links(
+                reader=reader,
+                ignore_unknown_nodes=ignore_unknown_nodes,
+                foreign_ids_reverse_mappings=foreign_ids_reverse_mappings,
+                ret_dict=ret_dict,
+                batch_size=batch_size,
+                # session=session
+            )
+
+            IMPORT_LOGGER.debug('STORING GROUP ELEMENTS...')
+            _import_groups(
+                group_count=reader.entity_count(GROUP_ENTITY_NAME),
+                group_uuids=reader.iter_group_uuids(),
+                foreign_ids_reverse_mappings=foreign_ids_reverse_mappings
+            )
+
+        ######################################
+        # Put everything in a specific group #
+        ######################################
+        # Note this is done in a separate transaction
+        group = _make_import_group(
+            group=group,
+            existing_entries=existing_entries,
+            new_entries=new_entries,
+            foreign_ids_reverse_mappings=foreign_ids_reverse_mappings
+        )
+
+    # Summarize import
+    result_summary(ret_dict, getattr(group, 'label', None))
+
+    return ret_dict
+
+
+def _generate_entity_data(
+    *, entity_name: str, reader: ArchiveReaderAbstract, new_entries: Dict[str, Dict[str, dict]],
```

Comment:
I think you are misunderstanding this `*`; it is not the same as `*args`, it enforces that all calls to the function must use keyword arguments for all the arguments after the `*`, i.e. `func(a=1, b=2)` instead of `func(1, 2)`.
Definitely start using it in your own code 😄 

(see below also)